symengine/printer.cpp

```diff
@@ -379,7 +379,10 @@ void StrPrinter::bvisit(const Mul &x)
             den++;
         } else {
             if (eq(*(p.second), *one)) {
```

Comment:
Change this to `eq(*(p.second), *one) and neq(*(p.first), *E)` and you won't need the changes below